### PR TITLE
Lint alignment of components with `forScopes`

### PR DESCRIPTION
**Pull request by ptgott**
The `forScopes` field of a navigation entry in `docs/config.json` determines which scope switcher buttons appear on a docs page. If a scoped component supports a scope that does not appear in the switcher, the user can only adjust the visibility of the component using the `scope` query string.

Further, if a `Tabs` component includes some scopes but not all of the ones listed in a page's `forScopes`, it's likely that the page's `forScopes` setting is misconfigured.

This change adds a linter that checks whether:

- A page's scoped components include any scopes that are _not_ listed in `forScopes`.
- `Tabs` components that include at least one scope in the `forScopes` configuration _also_ include the remaining scopes. The linter only performs this check for `Tabs` components since it's the only component that we expect to always include all scopes within a set.

This change will make it easier to incorporate Teleport Team instructions into the docs, since we can configure the linter to check for the `team` scope in scoped `Tabs` components and ensure that the `forScopes` setting is configured correctly.

### CODE REVIEW

**Comment by ptgott** [comment]
Converting to draft until I fix the `yarn typecheck` issues

**Review by benarent — APPROVED** [review]
It looks like the forScopes is the reason for https://github.com/gravitational/docs/pull/330 failing to pass, I manually removed team scopes from the 14 submodule and was able to build locally.

**Comment by ptgott** [comment]
Note: we should wait until v14 is out before we merge this. Doing so will simplify the process of getting this linter to pass, since v11 of the docs don't support Teleport Team.

**Comment by ptgott** [comment]
Closing because we are removing the scoped visibility functionality from the docs (#383).

**Pull request closed without merging**
